src/Core/Managed/Shared/Extensibility/AutocollectedMetricsExtractor.cs

```diff
@@ -252,6 +263,29 @@ private void ExtractMetrics(ITelemetry fromItem)
             }
         }
 
+        private bool EnsureItemNotSampled(ITelemetry item)
+        {
+            ISupportSampling potentiallySampledItem = item as ISupportSampling;
+
+            if (potentiallySampledItem != null && potentiallySampledItem.SamplingPercentage.HasValue)
```

Comment:
I'd recommend to check whether it's not 100-0.001. Just in case somebody wants to ensure the item will be send and set the sampling rate explicitly. Like:

``` csharp
var veryImportantEvent = new EventTelemetry("app is down") { SamplingPercentage = 100; }
client.TrackEvent(veryImportantEvent);
```